src/wayland/view-backend-wayland.cpp

```diff
@@ -181,29 +181,27 @@ ViewBackend::ViewBackend(struct wpe_view_backend* backend)
 
     m_surface = wl_compositor_create_surface(m_display.interfaces().compositor);
 
+    // In case that more than one protocol is available pick the first that matches.
     if (m_display.interfaces().xdg) {
         m_xdgSurface = xdg_shell_get_xdg_surface(m_display.interfaces().xdg, m_surface);
         xdg_surface_add_listener(m_xdgSurface, &g_xdgSurfaceListener, &m_resizingData);
         xdg_surface_set_title(m_xdgSurface, "WPE");
-    }
-
-    if (m_display.interfaces().xdg_v6) {
+    } else if (m_display.interfaces().xdg_v6) {
```

Comment:
If I understand correctly, the unversioned `xdg_shell` was (is?) the name used during development of the protocol, and the code should check first for the `zxdg_shell_v6` protocol *first* and use it if available, with `xdg_shell` as a fall-back for compositors which do not implement the versioned one.